### PR TITLE
Ip per task

### PR DESCRIPTION
**Pull request by diegooliveira**
It's common to some shops use ip per task using something like calico. Adding support for such configuration.

### CODE REVIEW

**Comment by emilevauge** [comment]
Thanks @diegooliveira! Could you rebase your PR to master?

**Comment by diegooliveira** [comment]
@emilevauge rebase done

**Comment by diegooliveira** [comment]
@emilevauge Tests passing.

**Comment by diegooliveira** [comment]
@emilevauge do you need any additional change?

File: provider/marathon_test.go [hunk 1]
```diff
@@ -674,10 +788,13 @@ func TestMarathonTaskFilter(t *testing.T) {
 	}
 
 	provider := &Marathon{}
-	for _, c := range cases {
+	for idx, c := range cases {
+		if idx == 4 {
```
Comment (emilevauge) [inline provider/marathon_test.go]: Not sure about that ;)

**Comment by emilevauge** [comment]
@diegooliveira Other than my comment, looks good to me :)
Thanks a lot !
/cc @containous/traefik

File: provider/marathon_test.go (hunk 1 above)
Comment (diegooliveira) [inline provider/marathon_test.go]: some debug code, fixed in the commit c3088d02f9bebf0272cd927d95cce506daec9930

**Pull request closed without merging**